src/content/blog/ga_qas.md

```diff
@@ -0,0 +1,232 @@
+---
+title: An Introduction to GA-QAS
+author: Vu Tuan Hai
+day: 21
+month: 5
+year: 2024
+tags:
+  - quantum architecture search
+  - python
+  - qiskit
+  - genetic-algorithm
+---
+
+In various quantum optimization problems, choosing the right ansatz is a critical point that will affect the accuracy of the result. Many template ansatzes have been proposed, such as Graph Ansatz and EffecientSU2, … but they are limited in their applicability. Hence, we've created a search engine called **GA-QAS** (Genetic Algorithm for Quantum Architecture Search) to aid in the discovery of the right ansatz. In this post, we will guide you on how to use it efficiently.
+
+
+## Now, let’s start!
+
+# Step 0. Setup the environment
+
+Make sure that you installed Python 3 and the following packages:
+
+- qiskit 1.0+
+- matplotlib
+- tdqm
+
+We will use a Python package named <a href="https://github.com/vutuanhai237/qoop">$\langle \mathrm{qo}|\mathrm{op} \rangle$</a>, which is a core package for developing quantum circuit optimization frameworks. You can download it via `git` as:
+
+```
+git clone https://github.com/vutuanhai237/qoop.git
+```
+
+Note that you should put it on the same level as your Python/Jupyter Notebook file. The best is:
+
+```
+folder_name
+│   your_python_file.py
+|   your_jupyter_file.ipynb
+└───qoop
+│   └───evolution
+│       │   environment.py
+│       │   cross_over.py
+│       │   ...
+```
+
+# Step 1. Define problem (W state preparation)
+
+<figure>
+    <img src='/images/ga-qas/ga-qas_scheme.png' width="600" style="margin: auto" alt='' />
+    <figcaption>
+    Fig 1. (a) State preparation scheme based on quantum compilation technique, where U, V are ansatz and state that need to prepare[1][2]. (b) Some default ansatz, such as hyper-graph.
+    </figcaption>
+</figure>
+
+First, we define the problem, take example quantum state preparation, such as W state preparation. A 3-qubit W state is defined as:
```

Comment:
Thanks for clearing this up a bit. I still think it needs more context for someone who might not understand that state prep is a vital part of QC. I made a recommendation here https://github.com/unitaryfund/unitary.fund/pull/483#discussion_r1657697366, but I think it should be extended slightly.

Sorry to go harp on this so much, but I think it's crucial to try to bring as many people in as possible and this is the the hook.

Reply:
As your recommendation, I add an introduction about quantum state preparation!

```suggestion
Quantum state preparation (QSP) is a process that
uses a trainable unitary operator to transform the initial state into a desired target state. This process is crucial for quantum computation and information processing. Example: in a quantum machine learning application, if you want to put an image into the quantum circuit for training, you must flat the image as a vector and then use QSP!

First, we define the problem, take example W - state preparation. A 3-qubit W state is defined as:

$$
|W_3\rangle = V|000\rangle = \frac{1}{\sqrt{3}}(|001\rangle + |010\rangle + |100\rangle)
$$

To prepare a W state (or other states), we will need a unitary operator $V$ generated systematically. Normally, $V$ requires large resources (such as large depth and number of gates). Therefore, we need to find another unitary $U(\theta)$ (trainable) that is approximate to $V$ but requires fewer resources than $V$.
```